src/parser.rs

```diff
@@ -1067,6 +1068,57 @@ impl<'a> Parser<'a> {
         }))
     }
 
+    pub fn parse_array_agg_expr(&mut self) -> Result<Expr, ParserError> {
+        self.expect_token(&Token::LParen)?;
+        let distinct = self.parse_all_or_distinct()?;
```

Comment:
`parse_all_or_distinct` will accept `ALL`, which isn't accepted by Snowflake and BigQuery, although it is by PostgreSQL [(3)].

I'd suggest you to add support for `ALL`, or adjust the structure to sinalize an `ALL`. Having this generates undesired permissiveness.

[1] : https://docs.snowflake.com/en/sql-reference/functions/array_agg.html
[2] : https://cloud.google.com/bigquery/docs/reference/standard-sql/aggregate_functions#array_agg
[3] : https://www.postgresql.org/docs/9.5/sql-expressions.html#SYNTAX-AGGREGATES